blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AbstractMetricsRW.java

```diff
@@ -148,22 +150,19 @@ protected int getTtl(Locator locator, RollupType rollupType, Granularity granula
      * @param timestampToRollupMap  a map of timestamp to rollup
      * @return
      */
-    protected <T extends Rollup> Points<T> convertToPoints(final Map<Long, T> timestampToRollupMap) {
-        Points<T> rollupPoints =  new Points<T>();
-        for (Map.Entry<Long, T> rollup : timestampToRollupMap.entrySet() ) {
-            rollupPoints.add(new Points.Point<T>(rollup.getKey(), rollup.getValue()));
+    protected <T extends Object> Points<T> convertToPoints(final Map<Long, T> timestampToRollupMap) {
```

Comment:
I agree it's kind of silly. But I don't like the fact that a lot of method signatures have to be changed from specific class to just ```Object```. For example, this method in EnumMetricData:
```
    private void populateEnumValueToCountMap(Map<Long, BluefloodEnumRollup> enumRollupMap, Map<Long, String> enumHashValuesMap)
```
had to be changed to:
```
    private void populateEnumValueToCountMap(Map<Long, Object> enumRollupMap, Map<Long, String> enumHashValuesMap) 
```
in Greg's first iteration of this PR.

It does not make sense to change that method to take a ```Map<Long, Object>```. It is a class about Enum.

The problem is he wants to use this method for simple numbers too, like Long, Float, Double. There is no way to achieve that because the Rollup objects don't share a common parent class with those Long, Float, Double.